src/vs/workbench/contrib/terminal/browser/terminalView.ts

```diff
@@ -217,7 +217,11 @@ export class TerminalViewPane extends ViewPane {
 	}
 
 	public focus() {
-		this._terminalService.getActiveTab()?.setWillFocus(true);
+		if (this._terminalService.getActiveInstance()) {
+			this._terminalService.getActiveInstance()?.focusWhenReady();
+		} else {
+			this._terminalService.willFocusNewTerminal = true;
+		}
```

Comment:
I was thinking more:

```ts
await this._terminalService.initialized; // Just returns _remoteTerminalsInitialized?
this._terminalService.getActiveInstance()?.focusWhenReady();
```

This seems preferable to focus-related props in TerminalService and TerminalTab. Alternatively:

```ts
public focus() {
  if (this._terminalService.connectionState === TerminalConnectionState.Connecting) {
    // Note that focus may be shifted unexpected here if this is called, then editor is focused manually, then connection happens
    // Could verify document.activeElement is the same to avoid that?
    this.register(this._terminalService.onDidChangeConnectionState(() => this._focus();
    return;
  }
  this._focus();
}

private _focus() {
  this._terminalService.getActiveInstance()?.focusWhenReady();
}
```